XenModel/XenAPI/XenObjectDownloader.cs

```diff
@@ -74,25 +65,18 @@ public static bool LegacyEventSystem(Session session)
         /// <param name="session">The session over which to download the objects. Must not be null.</param>
         /// <param name="changes">The queue that the ObjectChanges will be put into. Must not be null.</param>
         /// <param name="cancelled">Used by GetEvents().</param>
-        /// <param name="legacyEventSystem">True if legacy event system (event.next) should to be used.</param>
         /// <param name="token">Used by GetEvents().</param>
-        public static void GetAllObjects(Session session, LockFreeQueue<ObjectChange> changes, HTTP.FuncBool cancelled, bool legacyEventSystem, ref string token)
+        public static void GetAllObjects(Session session, LockFreeQueue<ObjectChange> changes, HTTP.FuncBool cancelled, ref string token)
         {
-            if (legacyEventSystem)
-            {
-                DownloadObjects(session, changes);
-                return;
-            }
-            
             // download objects that are not covered by event.from(), e.g. Roles
-            List<ObjectChange> list = new List<ObjectChange>();
-            Download_Role(session, list);
-            foreach (ObjectChange o in list)
-                changes.Enqueue(o);
+
+            var roleRecords = Role.get_all_records(session);

```

Comment:
Pleae note that I moved the content of the method here to avoid iterating twice through the objects.